Chatto/Source/ChatController/BaseChatViewController.swift

```diff
@@ -405,3 +406,38 @@ extension BaseChatViewController { // Rotation
         }, completion: nil)
     }
 }
+
+// MARK: - ContainerControllerProtocol
+extension BaseChatViewController: ContainerControllerProtocol {
+    open var contentContainerBottomMargin: CGFloat {
+        return self.inputContainerBottomConstraint.constant
+    }
+
+    open func changeContainerBottomMargin(withNewValue newValue: CGFloat, animated: Bool = false, callback: (() -> Void)? = nil) {
+        self.changeContainerBottomMargin(withNewValue: newValue, animated: animated, duration: CATransaction.animationDuration(), callback: callback)
+    }
+
+    open func changeContainerBottomMargin(withNewValue newValue: CGFloat, animated: Bool = false, duration: CFTimeInterval, initialSpringVelocity: CGFloat = 0.0, callback: (() -> Void)? = nil) {
+        guard self.inputContainerBottomConstraint.constant != newValue else { callback?(); return }
+        self.isAdjustingInputContainer = true
+        let layoutBlock = {
+            self.inputContainerBottomConstraint.constant = max(newValue, self.bottomLayoutGuide.length)
+            self.view.layoutIfNeeded()
+        }
+
+        if animated {
+            UIView.animate(withDuration: duration,
+                           delay: 0.0,
+                           usingSpringWithDamping: 1.0,
+                           initialSpringVelocity: initialSpringVelocity,
+                           options: .curveLinear,
+                           animations: layoutBlock) { (_) in
```

Comment:
It would be more readable to have `completion` parameter here. Now it looks like this block is the animations block.